azkaban-common/src/main/java/azkaban/executor/OnContainerizedExecutionEventListener.java

```diff
@@ -0,0 +1,86 @@
+package azkaban.executor;
+
+import azkaban.Constants;
+import azkaban.DispatchMethod;
+import azkaban.flow.Flow;
+import azkaban.flow.FlowUtils;
+import azkaban.project.Project;
+import azkaban.project.ProjectManager;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Singleton
+public class OnContainerizedExecutionEventListener implements OnExecutionEventListener{
+
+  private static final Logger logger =
+      LoggerFactory.getLogger(OnContainerizedExecutionEventListener.class);
+  private final ExecutorLoader executorLoader;
+  private final ExecutorManagerAdapter executorManagerAdapter;
+  private final ProjectManager projectManager;
+
+  @Inject
+  public OnContainerizedExecutionEventListener(ExecutorLoader executorLoader,
+      ExecutorManagerAdapter executorManagerAdapter, ProjectManager projectManager) {
+    this.executorLoader = executorLoader;
+    this.executorManagerAdapter = executorManagerAdapter;
+    this.projectManager = projectManager;
+  }
+
+  @Override
+  public void onExecutionEvent(final ExecutableFlow flow, final String action) {
+    if (action.equals("Restart Flow")) {
```

Comment:
Make "Restart Flow" a CONST string in Constants

Reply:
Done